### PR TITLE
stops creating ND for new cluster for BYO provider

### LINKED ISSUE #3159: Do not create an initial node deployment for "KubeAdm/BringYourOwn" provider
**Description** When a users selects `KubeAdm` provider for this cluster there is no need to create an initial Node Deployment for that cluster in the API server. This essentially means that there is no cloud provider and nodes will be added to the cluster manually.

**Acceptance criteria**
- The API doesn't create an initial ND for `KubeAdm` provider
- The POST `/api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments` EP doesn't create a ND for `KubeAdm`

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: there is no need to create a node deployment if the provider is KubeAdm as in this case the nodes are added manually by an admin.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3159 

**Does this PR introduce a user-facing change?**:

```release-note
the API will no longer attempt to create an initial node deployment for new kubeadm clusters
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@p0lyn0mial Can we have this reviewed to make sure it ends up in 2.10.1?

**Review by maciaszczykm — APPROVED** [review]
/lgtm
/approve

File: api/pkg/handler/v1/cluster/cluster.go [hunk 1]
```diff
@@ -80,6 +80,16 @@ func CreateEndpoint(sshKeyProvider provider.SSHKeyProvider, cloudProviders map[s
...
 		// Create the initial node deployment in the background.
 		if req.Body.NodeDeployment != nil {
+			// for BringYourOwn provider we don't create ND
+			isBYO, err := common.IsBringYourOwnProvider(spec.Cloud)
+			if err != nil {
+				return nil, errors.NewBadRequest("invalid spec: %v", err)
+			}
+			if isBYO {
+				glog.V(5).Infof("KubeAdm provider detected an initial node deployment won't be created for cluster %s", newCluster.Name)
+				return nil, nil
```
Comment (maciaszczykm) [inline api/pkg/handler/v1/cluster/cluster.go]: One thing that I am not completely sure is if we should return nil here if the cluster was created.

/hold

Comment (p0lyn0mial) [reply api/pkg/handler/v1/cluster/cluster.go]: I agree, we could return the cluster resource, I will fix it.

**Comment by p0lyn0mial** [comment]
@maciaszczykm i've updated the pull, PTAL.

**Comment by maciaszczykm** [comment]
/lgtm

Thanks :)

**Comment by p0lyn0mial** [comment]
/hold cancel

@maciaszczykm thx for reviewing :)

**Pull request merged**